File: Data Structures & Algorithms/min-cost-climbing-stairs/submission-2.py
class Solution:
    def minCostClimbingStairs(self, cost: List[int]) -> int:
        mc = [0, 0]
        for i in range(2, len(cost) + 1):
            mc.append(min(cost[i - 1] + mc[i - 1], cost[i - 2] + mc[i - 2]))
        return mc[-1]

File: Data Structures & Algorithms/max-water-container/submission-2.cpp
class Solution {
public:
    int maxArea(vector<int>& heights) {
        int l = 0, r = heights.size() - 1;
        int res = 0;
        while (l < r) {
            int curr = min(heights[l], heights[r]) * (r - l);
            res = max(res, curr);
            if (heights[l] < heights[r]) ++l;
            else --r;
        }
        return res;
    }
};
